src/NHibernate/Criterion/SQLProjection.cs

```diff
@@ -37,15 +39,48 @@ internal SQLProjection(string sql, string groupBy, string[] columnAliases, IType
 
 		public SqlString ToSqlString(ICriteria criteria, int loc, ICriteriaQuery criteriaQuery)
 		{
-			//SqlString result = new SqlString(criteriaQuery.GetSQLAlias(criteria));
-			//result.Replace(sql, "{alias}");
-			//return result;
-			return new SqlString(sql?.Replace("{alias}", criteriaQuery.GetSQLAlias(criteria)));
+			return GetSqlString(criteria, criteriaQuery, sql);
+		}
+
+		/// <summary>
+		/// Provide list of criteria aliases that's used in SQL projection.
+		/// To be replaced with SQL aliases.
+		/// </summary>
```

Comment:
I don't not really understand where it's going. If other PR is preferred - I see no point in members obsoletion. All those obsolete members were driven by breaking changes introduced in this PR.  And why are you suggesting to comment members that are not present in other version?

Reply:
Because it is not yet the end of it. If this PR if finally chosen, public API meant to be used by user code should be fully commented.

Reply:
About member obsoletion, the subject still exists, since `Expression` is commented as "semi-obsolete". At some point, something should be done about it, don't you think?

Reply:
Ok. But unless it's chosen I see no point on working on obosletion here. It's just easier to make those changes again - they won't merge nicely with other PR.